services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java

```diff
@@ -170,6 +165,9 @@ public TokenEndpoint(KeycloakSession session, TokenManager tokenManager, EventBu
         this.tokenManager = tokenManager;
         this.realm = session.getContext().getRealm();
         this.event = event;
+        this.request = session.getContext().getContextObject(HttpRequest.class);
+        this.httpResponse = session.getContext().getContextObject(HttpResponse.class);
```

Comment:
I wonder, should we use a dedicated methods for this? Or maybe in the future, when we create wrappers for `HttpResponse`, `HttpResponse`, etc.

Reply:
That is the idea. As mentioned in the description, this should be done in a follow-up PR. That method should return our own interface.